File: app/src/main/java/com/pavelpotapov/guessthecelebrity/service/SoundService.java
package com.pavelpotapov.guessthecelebrity.service;

import android.app.Service;
import android.content.Intent;
import android.content.res.AssetFileDescriptor;
import android.media.MediaPlayer;
import android.os.IBinder;

import androidx.annotation.Nullable;

import java.io.IOException;

public class SoundService extends Service implements MediaPlayer.OnPreparedListener {
    private static final String ACTION_PLAY = "play";
    private static final String ACTION_PAUSE = "pause";
    private static final String ACTION_RESUME = "resume";
    private static final String ACTION_STOP = "stop";
    private static final String ACTION_SWITCH = "switch";
    private MediaPlayer mediaPlayer;
    private AssetFileDescriptor afd;
    private int numberResumeCall = 0; // Колличество вызовов сетода onResume()
    private int position = 0; // Позиция на котором приостановилась мелодия
    private boolean volume = true; // Состояние звука (вкл/выкл)


    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        if (intent.getAction().equals(ACTION_PLAY)) {
            mediaPlayer = new MediaPlayer();
            try {
                afd = getApplicationContext().getAssets().openFd("sound_0.mp3");
                mediaPlayer.setDataSource(afd.getFileDescriptor(), afd.getStartOffset(), afd.getLength());
                mediaPlayer.setLooping(true);
                mediaPlayer.setOnPreparedListener(this);
                mediaPlayer.prepareAsync();
            } catch (IOException e) {
                e.printStackTrace();
            }
        } else if (intent.getAction().equals(ACTION_PAUSE)) {
            mediaPlayer.pause();
            position = mediaPlayer.getCurrentPosition();
        } else if (intent.getAction().equals(ACTION_RESUME)) {
            if (numberResumeCall == 0) {
                numberResumeCall++;
            } else {
                mediaPlayer.start();
            }
        } else if (intent.getAction().equals(ACTION_STOP)) {
            stopService(intent);
        } else if (intent.getAction().equals(ACTION_SWITCH)) {
            if (volume == true) {
                volumeOff();
                volume = false;
            } else if (volume == false) {
                volumeOn();
                volume = true;
            }
        }
        return START_STICKY;
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onPrepared(MediaPlayer mediaPlayer) {
        mediaPlayer.start();
    }

    @Override
    public boolean stopService(Intent name) {
        mediaPlayer.release();
        return super.stopService(name);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mediaPlayer.release();
    }

    private void volumeOff() {
        mediaPlayer.setVolume(0.0f, 0.0f);
    }

    private void volumeOn() {
        mediaPlayer.setVolume(1.0f, 1.0f);
    }
}

File: app/src/main/java/com/pavelpotapov/guessthecelebrity/utils/services/SoundService.kt
package com.pavelpotapov.guessthecelebrity.utils.services

import android.app.Service
import android.content.Intent
import android.media.MediaPlayer
import android.os.IBinder
import android.util.Log

private const val ACTION_PLAY: String = "ACTION_PLAY"
private const val ACTION_PAUSE: String = "ACTION_PAUSE"
private const val ACTION_RESUME: String = "ACTION_RESUME"
private const val ACTION_STOP: String = "ACTION_STOP"

private const val ACTION_SWITCH_SOUND: String = "ACTION_SWITCH_SOUND"

class SoundService : Service(), MediaPlayer.OnPreparedListener {
    private var player = MediaPlayer()
    private var position: Int = 0
    private var numberCallResume = 0
//    private val preferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
    private var volume = true
//    private var volume = preferences.getBoolean("volume", true)

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        when (intent?.action) {
            ACTION_PLAY -> {
                player.apply {
                    val afd = applicationContext.assets.openFd("sound_0.mp3")
                    player.setDataSource(afd.fileDescriptor, afd.startOffset, afd.length)
                    player.isLooping = true
                    player.setVolume(1.0f, 1.0f)
                    setOnPreparedListener(this@SoundService)
                    prepareAsync()
                }
            }
            ACTION_PAUSE -> {
                player.pause()
                position = player.currentPosition
                Log.d("player_position", position.toString())
            }
            ACTION_RESUME -> {
                if (numberCallResume == 0) {
                    Log.d("player_position", "Number of call resume() - $numberCallResume")
                    numberCallResume++
                } else {
                    player.seekTo(position)
                    Log.d("player_position", position.toString())
                    player.start()
                }
            }
            ACTION_STOP -> {
                stopService(intent)
            }
            ACTION_SWITCH_SOUND -> {
                if (volume) {
                    volumeOFF()
                    volume = false
                } else if (!volume) {
                    volumeON()
                    volume = true
                }
            }
        }
        return START_STICKY
    }

    override fun onBind(intent: Intent?): IBinder? {
        return null
    }

    override fun onPrepared(player: MediaPlayer) {
        Log.d("player_position", position.toString())
        player.start()
    }

    override fun stopService(name: Intent?): Boolean {
        player.release()
        return super.stopService(name)
    }

    override fun onDestroy() {
        super.onDestroy()
        player.release()
    }

    private fun volumeOFF() {
        player.setVolume(0.0f, 0.0f)
    }

    private fun volumeON() {
        player.setVolume(1.0f, 1.0f)
    }
}